### DISCUSSION TITLE
Logging queries from data sources based on configuration

### DISCUSSION THREAD

**Discussion by kishaningithub** [grafana/grafana#57263] [Data Sources]
To enable better debugging, it would be great if there was a way to enable query logging by data sources

From a code standpoint it would be as simple as setting `showSQL` to `true` in the xorm instance based on configuration which defaults to false around this below region of code

https://github.com/grafana/grafana/blob/e829b5014911363be862306b34a41fb0eeb13362/pkg/tsdb/sqleng/sql_engine.go#L151

just like how we have done the same for the db which stores grafana's internal state based on `database.log_queries` configuration

https://github.com/grafana/grafana/blob/e829b5014911363be862306b34a41fb0eeb13362/pkg/services/sqlstore/sqlstore.go#L401

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
